test/Parse/raw_string.swift

```diff
@@ -0,0 +1,120 @@
+// RUN: %target-swift-frontend -dump-ast %s 2>&1 | %FileCheck %s
+
+import Swift
+
+// ===---------- Multiline RawString --------===
+
+_ = ##"""
+    One
+    ""Alpha""
+    """##
+// CHECK: "One\n\"\"Alpha\"\""
+
+_ = ##"""
+    Two
+  Beta
+  """##
+// CHECK: "  Two\nBeta"
+
+_ = #"""
+    Three\r
+    Gamma\
+  """#
+// CHECK: "  Three\\r\n  Gamma\\"
+
+_ = ###"""
+    Four \(foo)
+    Delta
+"""###
+// CHECK: "    Four \\(foo)\n    Delta"
+
+_ = ##"""
+  print("""
+    Five\##n\##n\##nEpsilon
+    """)
+  """##
+// CHECK: "print(\"\"\"\n  Five\n\n\nEpsilon\n  \"\"\")"
+
+// ===---------- Single line --------===
+
+_ = #""Zeta""#
+// CHECK: "\"Zeta\""
+
+_ = #""Eta"\#n\#n\#n\#""#
+// CHECK: "\"Eta\"\n\n\n\""
+
+_ = #""Iota"\n\n\n\""#
+// CHECK: "\"Iota\"\\n\\n\\n\\\""
+
+_ = #"a raw string with \" in it"#
+// CHECK: "a raw string with \\\" in it"
+
+_ = ##"""
+      a raw string with """ in it
+      """##
+// CHECK: "a raw string with \"\"\" in it"
+
+let foo = "Interpolation"
+_ = #"\b\b \#(foo)\#(foo) Kappa"#
+// CHECK: "\\b\\b "
+// CHECK: " Kappa"
+
+_ = """
+  interpolating \(##"""
+    delimited \##("string")\#n\##n
+    """##)
+  """
+
+// CHECK: "interpolating "
+// CHECK: "delimited "
+// CHECK: "string"
+// CHECK: "\\#n\n"
+
+#"unused literal"#
+// CHECK: "unused literal"
+
+// ===---------- From proposal --------===
+
+_ = #"This is a string"#
+// CHECK: "This is a string"
+
+_ = #####"This is a string"#####
+// CHECK: "This is a string"
+
+_ = #"enum\s+.+\{.*case\s+[:upper:]"#
+// CHECK: "enum\\s+.+\\{.*case\\s+[:upper:]"
+
+_ = #"Alice: "How long is forever?" White Rabbit: "Sometimes, just one second.""#
+// CHECK: "Alice: \"How long is forever?\" White Rabbit: \"Sometimes, just one second.\""
+
+_ = #"\#\#1"#
+/// CHECK: "\\#1"
+
+_ = ##"\#1"##
+/// CHECK: "\\#1"
+
+_ = #"c:\windows\system32"#
+/// CHCECK: "c:\\windows\\system32"
```

Comment:
Typo: `CHCECK` => `CHECK`

Does it matter how many forward slash characters are used?
i.e. `/// CHECK:` versus `// CHECK:`

Would the following also be a valid raw string test?

```
_ = #"""
###################################################################
## This source file is part of the Swift.org open source project ##
###################################################################
"""#
```